x-pack/test/apm_api_integration/trial/tests/services/transaction_groups_charts.ts

```diff
@@ -0,0 +1,150 @@
+
+        describe('without uiFilters', () => {
+          before(async () => {
+            response = await supertest.get(
+              `/api/apm/services/${serviceName}/transaction_groups/charts?start=${start}&end=${end}&transactionType=${transactionType}&`
+            );
+          });
+          it('should return an error response', () => {
+            expect(response.status).to.eql(400);
+          });
+        });
+
+        describe('with empty uiFilters', () => {
+          const uiFilters = encodeURIComponent(JSON.stringify({}));
+          before(async () => {
+            response = await supertest.get(
+              `/api/apm/services/${serviceName}/transaction_groups/charts?start=${start}&end=${end}&transactionType=${transactionType}&uiFilters=${uiFilters}`
+            );
+          });
+
+          it('should have a successful response', () => {
+            expect(response.status).to.eql(200);
+          });
+
+          it('should include an ML job id', () => {
+            expect(response.body).to.have.property('anomalyTimeseries');
+            expect(response.body.anomalyTimeseries).to.have.property('jobId');
+            expectSnapshot(response.body.anomalyTimeseries.jobId).toMatchInline(
+              `"apm-environment_not_defined-7ed6-high_mean_transaction_duration"`
+            );
+          });
+
+          it('should return the correct anomaly boundaries', () => {
+            expect(response.body).to.have.property('anomalyTimeseries');
+            expect(response.body.anomalyTimeseries).to.have.property('anomalyBoundaries');
+            expect(response.body.anomalyTimeseries.anomalyBoundaries.length).to.equal(0);
+          });
+        });
+
+        describe('with uiFilters environment defined', () => {
+          const uiFilters = encodeURIComponent(JSON.stringify({ environment: 'production' }));
+          before(async () => {
+            response = await supertest.get(
+              `/api/apm/services/${serviceName}/transaction_groups/charts?start=${start}&end=${end}&transactionType=${transactionType}&uiFilters=${uiFilters}`
+            );
+          });
+
+          it('should have a successful response', () => {
+            expect(response.status).to.eql(200);
+          });
+
+          it('should include an ML job id', () => {
+            expect(response.body).to.have.property('anomalyTimeseries');
+            expect(response.body.anomalyTimeseries).to.have.property('jobId');
+            expectSnapshot(response.body.anomalyTimeseries.jobId).toMatchInline(
+              `"apm-production-229a-high_mean_transaction_duration"`
+            );
+          });
+
+          it('should return the correct anomaly boundaries', () => {
+            expect(response.body).to.have.property('anomalyTimeseries');
+            expect(response.body.anomalyTimeseries).to.have.property('anomalyBoundaries');
+            expect(response.body.anomalyTimeseries.anomalyBoundaries.length).to.be.greaterThan(0);
```

Comment:
What is the value of checking if a property `anomalyBoundaries` exists if we afterwards assert the length of the same property? Wouldn't the second expect should be sufficient?

Reply:
It's possible to have `anomalyBoundaries` as an empty array `[]` but we want to make sure it has data. The extra check for the nested property was to improve error messaging on a failure: `Error: expected object to have a property 'anomalyTimeseries` vs `TypeError: Cannot read property 'anomalyBoundaries' of undefined`.

Reply:
If `anomalyBoundaries` is optional we can fix it like:
```
            expect(response.body.anomalyTimeseries.anomalyBoundaries?.length).to.be.greaterThan(0);
```

But I don't think it is, is it?

Reply:
`anomalyTimeseries` is optional, but if it's there, then `anomalyTimeseries` should be a nested property. so we could safely eliminate that nested property check.